Demo/Demo/Card/CardFormatter.swift

```diff
@@ -0,0 +1,47 @@
+class CardFormatter {
+
+    func formatCardNumber(_ newCardNumber: String, previousCardNumber: String = "") -> String {
+        if newCardNumber.count < previousCardNumber.count {
+            return newCardNumber
+        }
+
+        var cardString: String = newCardNumber.replacingOccurrences(of: " ", with: "")
+
+        let cardType = CardType.unknown.getCardType(cardString)
+
+        if previousCardNumber.count == cardType.maxLength + cardType.cardNumberIndices.count {
+            return previousCardNumber
+        } else {
```

Comment:
this is getting into the nitpicking/edge case domain, but this might not work if user copy/paste in a new string with this length (we'll skip formatting the new string in this case :smile:)
maybe we can remove this and just leave the `else` part. What do you think?

Reply:
Sure! I think that makes sense!